components/misc/rng.hpp

```diff
@@ -3,6 +3,7 @@
 
 #include <cassert>
 #include <random>
+#include <chrono>
```

Comment:
A bunch of stuff depends on rng.hpp and including chrono in it is something scrawl advised against.

Reply:
Ok, I will hide default seed generation into a function and will place definition into .cpp file.

Reply:
@Capostrophic is this OK?